db/migrate/20150113100404_add_osfamilies.rb

```diff
@@ -0,0 +1,59 @@
+# This migration creates all the necessary schema models to ensure flavor and appliance type
+# differentiation into OS families (typically Linux and Windows). This is necessary to facilitate
+# proper billing where the price of a flavor depends on the installed OS.
+
+class AddOsfamilies < ActiveRecord::Migration
+  def up
+    create_table :atmosphere_os_families do |t|
+      t.string :os_family_name, null:false, default: "Windows"
+    end
+
+    create_table :atmosphere_virtual_machine_flavor_os_families do |t|
```

Comment:
Maybe `FlavorOsFamily`? The name `VirtualMachineFlavorOsFamily` is soooooooo long and all relation names are sooooooo long.